### PR TITLE
Add Test#get_signal()

### LINKED ISSUE #34526: Provide AbortSignal from test object?
```js
promise_test(t => {
  const controller = new AbortController();
  const { signal } = controller;
  t.add_cleanup(() => controller.abort());

  document.body.addEventListener("pointerdown", console.log, { signal });
});
```

```js
promise_test(t => {
  const { signal } = t;
  
  document.body.addEventListener("pointerdown", console.log, { signal });
});
```

What do you think? This way the `signal` getter should throw if there's no AbortSignal support.

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #34526

### CODE REVIEW

**Comment by jgraham** [comment]
So, part from the CI failures, this requires two things: 
* An [RFC](https://github.com/web-platform-tests/rfcs), since it's adding new API surface to testharness.js. See the other examples there, but that should describe the API, the use cases, and any tradeoffs that went into the proposed design.
* Full try runs by pushing the branch to at least `triggers/firefox_nightly` and `triggers/chrome_dev`, to ensure that the testharness changes don't unexpectedly regress any other tests.

**Review by jgraham — DISMISSED** [review]
Marking as "Request Changes" until we have an RFC to compare against.

However in general this looks pretty straightforward so my initial feeling is that there aren't going to be major problems with this proposal.

**Comment by saschanaz** [comment]
@jgraham or @foolip, can either of you approve this given the RFC is merged?

File: resources/testharness.js [hunk 1]
```diff
@@ -2478,6 +2478,10 @@
         this._user_defined_cleanup_count = 0;
         this._done_callbacks = [];
 
+        if (typeof AbortController) {
```
Comment (foolip) [inline resources/testharness.js:2481]: Won't this always be true, since the string "undefined" is truthy?

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
Oh wait, now there are infrastructure/ test failures, due to "UNEXPECTED-OK /infrastructure/server/context.any.sharedworker-module.html"

 @saschanaz is there any chance that's because of these changes?

**Comment by saschanaz** [comment]
That didn't work, but looking at https://github.com/web-platform-tests/wpt/blob/master/infrastructure/server/context.any.js I doubt this can affect it 🤔

**Comment by saschanaz** [comment]
(@jgraham's req blocks the merge)

**Comment by foolip** [comment]
I'm hoping that https://github.com/web-platform-tests/wpt/pull/38545 will fix this.

**Comment by foolip** [comment]
The remaining failures is the same as in https://github.com/web-platform-tests/wpt/pull/28759 so I'll merge this.

**Pull request merged**